packages/ui/src/Avatar.tsx

```diff
@@ -135,56 +181,134 @@ export const Avatar = (props: AvatarProps): React.ReactElement => {
+        </Box>
+      );
+    } else if (editAvatarImage && Platform.OS !== "web" && !props.status) {
+      return (
+        <Box
+          bottom
+          paddingX={sizeIconPadding[size]}
+          position="absolute"
+          right
+          zIndex={5}
+          onClick={pickImage}
         >
-          <Text style={{fontSize, color: props.textColor ?? Unifier.theme.darkGray}}>
-            {computedInitials}
-          </Text>
-        </View>
-      )}
+          <Icon color="black" name="edit" size={size} />
+        </Box>
+      );
+    }
+    return null;
+  };
+
+  const renderStatusIcon = () => {
+    if (!props.status) {
+      return null;
+    }
+    // eslint-disable-next-line prefer-const
+    let {icon, color} = statusIcons[props.status];
+    if (
+      props.statusMobile &&
+      ["online", "away", "offline", "doNotDisturb"].includes(props.status)
+    ) {
+      icon = "mobile-alt";
+    }
+    if (!icon || !color) {
+      console.warn(`Avatar: Invalid status ${props.status}`);
+      return null;
+    }
+    return (
+      <Box bottom paddingX={sizeIconPadding[size]} position="absolute" right zIndex={5}>
+        <Icon color={color} name={icon} size={size} />
+      </Box>
+    );
+  };
+
+  const avatar = (
+    <Box height={height} position="relative" width={width}>
+      <Box
+        border={outline ? "white" : undefined}
+        height={height}
+        overflow="hidden"
+        position="relative"
+        rounding="circle"
+        width={width}
+        onHoverEnd={() => setHovered(false)}
+        onHoverStart={() => setHovered(true)}
+      >
+        {src && isImageLoaded ? (
+          // TODO: Make our Image component rounding work so that we can use it for Avatar. Currently it creates an
+          // unrounded box around the Image.
+          <Image
+            resizeMode={imageFit as ImageResizeMode}
+            source={{uri: src, cache: "force-cache"}}
+            style={{
+              borderRadius: radius,
+              height,
+              width,
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              overflow: "hidden",
+            }}
+            onError={handleImageError}
+          />
+        ) : (
+          <View
+            style={{
+              height,
+              width,
+              borderRadius: radius,
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              backgroundColor: props.backgroundColor
+                ? Unifier.theme[props.backgroundColor]
+                : Unifier.theme.gray,
+            }}
+          >
+            <Text style={{fontSize, color: props.textColor ?? Unifier.theme.darkGray}}>
+              {computedInitials}
+            </Text>
+          </View>
+        )}
+        {/* Needs to come after the image so it renders on top. */}
+        {renderEditIcon()}
```

Comment:
Do we expect to need the edit icon and the status icon simultaneously? The edit icon is currently cut off by the  circle portrait of the image. If we move it to the same level as the status icon, it should show correctly but only if we don't ever need both of them at the same time. Another option would be to move one both icons to the same level and shift one of the icons to the top right so they never clash

Reply:
Ah yeah, I fixed that by moving it to the middle and on hover only (pro is it's more similar to a lot of edit interfaces, con is we lose out on some discoverability that it's editable, but I think this a pretty normal pattern. open to pushback there though). Added a story for both, seems to work well.

Reply:
Sorry, I wasn't clear but I'm a fan of the hover in web. It's just mobile where the edit is cut off and where we might want to position the two icons differently. Just hoping to pull the edit up in the DOM one more level so that it's not cut off by the circle container of the image.

Reply:
Ohhh right! Yeah I'll see what I can do!